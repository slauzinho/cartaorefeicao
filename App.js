import React from "react";
import { StyleSheet } from "react-native";
import { createStackNavigator, createSwitchNavigator } from "react-navigation";
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import createSagaMiddleware from 'redux-saga'
import Home from "./components/Home";
import InstructionsScreen from "./screens/InstructionsScreen";
import ModalScreen from "./screens/ModalScreen";
import LoadingScreen from "./screens/LoadingScreen";
import NoCardScreen from "./screens/NoCardScreen";
import AddCardScreen from "./screens/AddCardScreen";
import reducer from './reducers';
import {rootSaga} from './rootSaga';

const sagaMiddleware = createSagaMiddleware()
const store = createStore(
  reducer,
  applyMiddleware(sagaMiddleware)
)

sagaMiddleware.run(rootSaga);

console.log(store.getState())

const MainStack = createStackNavigator(
  {
    Home: {
      screen: Home,
      headerMode: "none",
      header: null,
      navigationOptions: {
        header: null
      }
    },
    Add: AddCardScreen,
    Help: InstructionsScreen
  },
  {
    initialRouteName: "Home",
    headerMode: "screen",
    cardStyle: {
      backgroundColor: "white",
      flex: 1,
      flexDirection: "row",
      justifyContent: "center"
    }
  }
);

const NoCardStack = createStackNavigator(
  {
    Home: {
      screen: NoCardScreen,
      headerMode: "none",
      header: null,
      navigationOptions: {
        header: null
      }
    },
    Add: AddCardScreen,
    Help: InstructionsScreen
  },
  {
    initialRouteName: "Home",
    headerMode: "screen",
    cardStyle: {
      backgroundColor: "white",
      flex: 1,
      flexDirection: "row",
      justifyContent: "center"
    }
  }
);

const RootStack = createStackNavigator(
  {
    Main: {
      screen: MainStack
    },
    MyModal: {
      screen: ModalScreen
    }
  },
  {
    mode: "modal",
    headerMode: "none"
  }
);

const RootSwitch = createSwitchNavigator(
  {
    Loading: LoadingScreen,
    App: RootStack,
    EmptyState: NoCardStack
  },
  {
    initialRouteName: "Loading"
  }
);

const App = () => <Provider store={store}><RootSwitch style={styles.container} /></Provider>

export default App;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "center"
  }
});
