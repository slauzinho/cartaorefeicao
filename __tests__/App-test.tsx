import React from 'react';
import { render } from 'react-native-testing-library';
import App from '../App';

describe('App', () => {
  beforeEach(() => {
    jest.spyOn(console, 'error').mockImplementation(() => {});
  });
  it('can render the app', () => {
    const { toJSON } = render(<App />);
    expect(toJSON()).toMatchSnapshot();
  });
});
