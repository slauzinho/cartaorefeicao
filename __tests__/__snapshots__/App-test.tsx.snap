// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App can render the app 1`] = `
<ActivityIndicator
  animating={true}
  color="grey"
  hidesWhenStopped={true}
  size="large"
/>
`;
